File: agnese.py
boolD = True

def main(stringa):
    length = len(stringa)
    word = stringa
    answer = False
    x = 0
    while (answer == False and x < length):
        letter = word[x]
        position = x
        originPosition = x
        for y in range(0,len(word)):
            check = controlIfTheStringExistAfterThat(letter,word,position,originPosition)
            if (check == True):
                answer = True
            newString = word[position+1:len(word)]
            checkIfTheLetterExist = newString.find(letter)
            if (checkIfTheLetterExist>-1 and position+1!=len(word)):#//FIXME: deve controllare anche le parole dietro quando serve altrimenti salta
                position = position+1
                letter = letter+word[position]
            else:
                break
        x+=1    
    print(answer)
    
def controlIfTheStringExistAfterThat(letter,word,positions,originPosition):
    lentgh = len(letter)
    lentghSub1 = positions+lentgh -1
    lentghSub0 = ""
    sub0 = ""
    lentghSub2 = positions+lentgh+lentgh -1
    existElement = checkIfTheElementExist(letter,word,originPosition)

    if (lentgh == 1):
        lentghSub1 = positions+lentgh 
        lentghSub2 = positions+lentgh+lentgh  
    if (existElement):
        lentghSub0 = originPosition - lentgh
        sub0 = word[lentghSub0:originPosition]
    sub1 = word[lentghSub1:lentghSub2]
    sub2 = word[lentghSub2:lentghSub2+lentgh]
    answer = False
    if (letter == sub1 and sub1 != sub2 and letter != sub0 ):
        answer = True
    if (letter == sub2 and letter == sub1 and sub1 == sub2):
        answer = False
    if (answer == True):
        print(letter+letter)
    return answer


def checkIfTheElementExist(letter,word,positions):
    lentgh = len(letter)
    risp = False
    if (positions - lentgh >= 0):
        risp = True
    return risp




if __name__ == "__main__":
    if boolD:
        print("START")
    #canecane
    string = "piupiu"
    main(string)
    
    
    if boolD:
        print("STOP")

File: agnese.cpp

using namespace std;
#include <iostream>

bool controllIfAEllementExistBehindTheWord(string word, string char_, int index){
  bool answer = true;
  int len = char_.length();
  string sub0 = "";
  if ((index - len) >= 0){
    sub0 = word.substr((index - len),len);
  
  if (char_ == sub0 && "" != sub0){ 
    answer = false;
  }
  }
  
  return answer;
}


bool controllingHowMuchTimeExistAString(string word,string char_){
  
  int len = char_.length();
  string sub1;
  string sub2;
  string sub;
  bool answer = false;
    try
{
  sub = word.substr(len,word.length());
}
catch( std::out_of_range& exception )
{
  sub = "";
}
  try
{
  sub1 =   sub.substr(0,len);
}
catch( std::out_of_range& exception )
{
  sub1 =   sub.substr(0,word.length());
}
    try
{
    sub = sub.substr(len,len);
  sub2 = sub.substr(0,len);
}
catch( std::out_of_range& exception )
{
  sub2 = sub.substr(0,word.length());
}

  
  if (sub1 == char_ && sub2 != sub1){
    answer = true;
  }
  
  return answer;
}



bool controllingIfExistADouble(int index,string word){
  
  bool answer = false;
  int len = word.length();
  string substring = word.substr(index,word.length());
  int indexWord = 0;
  string char_ = ""+word;
  char_ = word[indexWord];
  int index_;
  bool answerForUndenstandHowManyElementAreInTheWord;
  bool ControllIfAEllementExistBehindTheWord ;
  while (answer==false && indexWord <= substring.length()){
    
    answerForUndenstandHowManyElementAreInTheWord = controllingHowMuchTimeExistAString(substring,char_);
    ControllIfAEllementExistBehindTheWord= controllIfAEllementExistBehindTheWord( word, char_, indexWord+index);
    if (answerForUndenstandHowManyElementAreInTheWord == true && ControllIfAEllementExistBehindTheWord == true){
      answer = true;
    }
    indexWord++;
    char_ = char_+substring[indexWord];
    if (word.find(char_)>100){
       break;
    }
    
  }
  return answer;
}




int main() {
  string character = "piupiu";
  bool answer = false;
  int index = 0;
  
  while (answer==false && index < character.length()){
    answer = controllingIfExistADouble(index,character);
    index++;
  };
  cout << answer << "\n";
  if (answer){
    
  }
  return 0;
}